src/components/Form.js

```diff
@@ -2,37 +2,48 @@ import React, { useState } from "react";
 import axios from "axios";
 import swal from "sweetalert";
 
-import { formData } from "../db/form";
+import { messageSend } from "../db/form";
 
 import "../styles/components/form.scss";
 
-const Form = ({ data: { title, paragraphs, textarea, link }, emailTitle }) => {
-    const [name, setName] = useState("");
-    const [email, setEmail] = useState("");
-    const [message, setMessage] = useState("");
+const Form = ({
+    data: { title, paragraphs, textareaPlaceholder, link },
+    emailTitle,
+}) => {
+    const [isSubmitting, setIsSubmiting] = useState(false);
 
-    const formReset = () => {
-        setName("");
-        setEmail("");
-        setMessage("");
+    const successfullySent = () => {
+        swal(...Object.values(messageSend.success));
     };
 
-    const handleSubmit = (e) => {
+    const failureSent = () => {
+        swal(...Object.values(messageSend.fail));
+    };
+
+    const handleSubmit = async (e) => {
         e.preventDefault();
-        axios({
-            method: "POST",
-            // TODO: url do zmiany
-            url: "http://localhost:3002/access",
-            data: { emailTitle, name, email, message },
-        }).then((response) => {
-            const result = response.data.status;
-            swal(
-                formData[result].title,
-                formData[result].message,
-                formData[result].icon
+        if (isSubmitting) return;
+        setIsSubmiting(true);
+
+        const { name, email, message } = e.target.elements;
+        try {
+            await axios.post(
+                // TODO: url do zmiany
+                "http://localhost:3002/access",
+                {
+                    emailTitle,
+                    name: name.value,
+                    email: email.value,
+                    message: message.value,
+                }
             );
-            result !== "fail" && formReset();
-        });
+            successfullySent();
+            e.target.reset();
+        } catch (err) {
+            failureSent();
+        } finally {
+            setIsSubmiting(false);
+        }
```

Comment:
You suppose to separate API's logic to a new folder 'api'

```
export const createNewMail = async (mainForm) => {
  const { data } = await axios.post(API_URL, form)
  return data
}
```

+ try and catch.
Implement that method to your component using React.useCallback

```
  const onSubmit = React.useCallback(async () => {
    const result = await createNewMail(mail)
    onMailCreated(result)
    setNewMail((prev) => ({ ...prev, name: '' " ......}))
  }, [onMailhCreated, mail])

```